### PR TITLE
Updates for recent versions of PHP

### PR DESCRIPTION
**Pull request by sagepe**
This updates the PHP code in such a way that it should be compatible up to at least PHP 7.4.

At the time of opening, the `facebookphp4/` subdirectory is still present, although I think this can be removed as I'm not aware of anything using it - but it would be good to get this confirmed.

Note that this was branched from the mapit-query-string branch, so includes the commits in #80.

### CODE REVIEW

**Review by gbp — COMMENTED** [review]
I haven't run but from my out of date PHP knowledge this all looks correct to me. Nice that they've replaced `create_function` with anonymous functions now.

File: phplib/admin-ratty.php [hunk 1]
```diff
@@ -230,7 +230,7 @@ function display($self_link) {
                     list($field_name, $field_description, $field_examples) = $row;
                     print "<b>$field_name:</b> $field_description. ";
                     print "e.g. " . implode(", ", array_map(
-                    create_function('$a', 'return "\'".trim_characters($a, 0, 50)."\'";'), $field_examples));
+                    function($a) { return "\'".trim_characters($a, 0, 50)."\'"; }, $field_examples));
```
Comment (dracos) [inline phplib/admin-ratty.php:233]: You need to not escape the single quotes any more.

**Review by dracos — CHANGES_REQUESTED** [review]
facebookphp4 was only ever used by PledgeBank.
One thing to fix I think.

File: phplib/admin-ratty.php (hunk 1 above)
Comment (sagepe) [reply phplib/admin-ratty.php]: I've pushed a fixup commit, assuming that's all good I'll squash it in before merging.

**Review by dracos — APPROVED** [review]

**Pull request merged**